src/components/InputSearch/InputSearch.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
+
+import classNames from 'classnames';
+
+import scss from './InputSearch.module.scss';
+
+import alertIcon from './assets/alertIcon.svg';
+import leftIcon from './assets/leftIcon.svg';
+import rightIcon from './assets/rightIcon.svg';
+
+import { IInputProps } from './IInputProps.types';
+
+export default function InputSearch(props: IInputProps) {
+  const labelClass = [scss.label];
+  const inputClass = [scss.input];
+  const legendClass = [scss.legend];
+  const fieldsetClass = [scss.fieldset];
+  const containerClass = [scss.container];
+  const [isFocused, setIsFocused] = useState(false);
+
+  const [value, setValue] = useState('');
+
+  const IconAlert = () => {
+    return <img className={scss.icon} src={alertIcon} />;
+  };
+
+  const IconRight = () => {
+    return <img src={rightIcon} />;
+  };
+
+  const handleClear = () => {
+    if (props.handleClear) props.handleClear();
+    setValue('');
+  };
+
+  const renderErrorMessage = () => (
+    <div className={scss.errorWrapper}>
+      <span className={scss.errorMessage}>{props.errorMessage}</span>
+    </div>
+  );
+
+  const onInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (props.onFocus) props.onFocus(event);
+    setIsFocused(true);
+  };
+
+  const onInputBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    setIsFocused(e.target.value.length !== 0);
+  };
+
+  if (
+    isFocused ||
+    (typeof props.value === 'string' && props.value.length > 0)
+  ) {
+    fieldsetClass.push(scss.fieldsetFocus);
+    legendClass.push(scss.legendFocus);
+    labelClass.push(scss.labelAnimate);
+    containerClass.push(scss.containerFocus);
+  }
+
+  if (props.errors) {
+    fieldsetClass.push(scss.fieldsetError);
+    containerClass.push(scss.containerError);
+    labelClass.push(scss.labelError);
+  }
```

Comment:
I usually don't care about `useMemo` usage, but if you think about this whole component, every time the user types something, a re-render is done, and because this is not inside a re-render, it means that this will be calculated each render (every time the user time), this will make you use more CPU that you want.